### PR TITLE
Core: QUnit.equiv object methods comparison

### LINKED ISSUE #914: QUnit.equiv ignores object methods
https://jsfiddle.net/xcrh9n3d/

Since we verify inherited properties anyway, we should replace the weird constructor-based "function" callback with a simple `return false` (relying solely on the strict equality check in `innerEquiv`).

### PR DESCRIPTION
**Pull request by kumarmj**
fixes #914

### CODE REVIEW

File: test/main/deepEqual.js [hunk 1]
```diff
@@ -1301,12 +1301,24 @@ QUnit.test( "Instances", function( assert ) {
...
+
 	assert.equal( QUnit.equiv( a1, a2 ), true, "Same property, same constructor" );
 
 	// b1.fn and b2.fn are functions but they are different references
 	// But we decided to skip function for instances.
 	assert.equal( QUnit.equiv( b1, b2 ), true, "Same property, same constructor" );
 
+	// Added as bugfix for same property, different reference functions.
+	assert.equal( QUnit.equiv( c1, c2 ), false, "Same property differnent reference, but different functions" );
+	assert.equal( QUnit.equiv( c1, c3 ), true, "Same property different reference, same functions" );
```
Comment (trentmwillis) [inline test/main/deepEqual.js]: I would prefer the assertion messages be formatted like such:

* "Same property with different reference and different function"
* "Same property with different reference but same function"

I think it makes it a bit clearer.

File: test/main/deepEqual.js [hunk 2]
```diff
@@ -1301,12 +1301,24 @@ QUnit.test( "Instances", function( assert ) {
...
+	c2 = new C( function() { return false; } );
+	c3 = new C( function() { return true; } );
+
 	assert.equal( QUnit.equiv( a1, a2 ), true, "Same property, same constructor" );
 
 	// b1.fn and b2.fn are functions but they are different references
 	// But we decided to skip function for instances.
 	assert.equal( QUnit.equiv( b1, b2 ), true, "Same property, same constructor" );
 
+	// Added as bugfix for same property, different reference functions.
```
Comment (trentmwillis) [inline test/main/deepEqual.js]: I think this comment is unnecessary. The assertion messages are enough.

**Comment by gibson042** [comment]
LGTM. @trentmwillis?

**Comment by trentmwillis** [comment]
Thanks @kumarmj!

**Pull request merged**